classes/blocks/class-base-block.php

```diff
@@ -235,4 +235,24 @@ public static function render_frontend( $attributes ) {
 
 		return '<div data-render="' . self::get_full_block_name() . '" data-attributes="' . htmlspecialchars( $json ) . '"></div>';
 	}
+
+	/**
+	 * Hydrate a `div` with the block's attributes in a `data-` attribute to be handled in the frontend.
+	 *
+	 * @param array $attributes The block's attributes.
+	 * @param array $content The block's content.
+	 */
+	public static function hydrate_frontend( $attributes, $content ) {
+		$json = wp_json_encode( [ 'attributes' => $attributes ] );
+
+		// Parse to get the only the block content and not the whole block.
```

Comment:
@mleray FYI, this new piece of code solves [this](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/932#issuecomment-1232542738).